contrib/client-with-cli-config/main.go

```diff
@@ -0,0 +1,85 @@
+// demonstration of using REST API client with cli config (push/pull with credential)
```

Comment:
Maybe this could be a in a gist or a separate example repo that we can link to from docs?